### PR TITLE
Update container-builder to CentOS stream 9

### PR DESCRIPTION
**Pull request by thom-at-redhat**

### CODE REVIEW

**Comment by thenets** [comment]
I got an "image not found error". Investigating it...
```
$ make container-image-builder
$(./receptor-tester.sh container-runtime) build \
        -t ansible/receptor:builder \
        -f ./environments/container-builder/Containerfile \
        ./../
[+] Building 1.4s (2/2) FINISHED                                                                                                                   docker:default
 => [internal] load build definition from Containerfile                                                                                                      0.1s
 => => transferring dockerfile: 1.37kB                                                                                                                       0.0s
 => ERROR [internal] load metadata for docker.io/library/centos:stream9                                                                                      1.2s
------
 > [internal] load metadata for docker.io/library/centos:stream9:
------
Containerfile:1
--------------------
   1 | >>> FROM centos:stream9
   2 |     
   3 |     # Python options = [3.9, 3.11, 3.12]
--------------------
ERROR: failed to solve: centos:stream9: failed to resolve source metadata for docker.io/library/centos:stream9: docker.io/library/centos:stream9: not found
```

File: tests/environments/container-builder/Containerfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
-FROM centos:8
+FROM centos:stream9
```
Comment (thenets) [inline tests/environments/container-builder/Containerfile]: I think we should consume the image explicitly from quay:
```quay.io/centos/centos:stream9```

Comment (thenets) [reply tests/environments/container-builder/Containerfile]: With this change, I was able to build the image.

**Comment by thom-at-redhat** [comment]
> I got an "image not found error". Investigating it...
> 
> ```
> $ make container-image-builder
> $(./receptor-tester.sh container-runtime) build \
>         -t ansible/receptor:builder \
>         -f ./environments/container-builder/Containerfile \
>         ./../
> [+] Building 1.4s (2/2) FINISHED                                                                                                                   docker:default
>  => [internal] load build definition from Containerfile                                                                                                      0.1s
>  => => transferring dockerfile: 1.37kB                                                                                                                       0.0s
>  => ERROR [internal] load metadata for docker.io/library/centos:stream9                                                                                      1.2s
> ------
>  > [internal] load metadata for docker.io/library/centos:stream9:
> ------
> Containerfile:1
> --------------------
>    1 | >>> FROM centos:stream9
>    2 |     
>    3 |     # Python options = [3.9, 3.11, 3.12]
> --------------------
> ERROR: failed to solve: centos:stream9: failed to resolve source metadata for docker.io/library/centos:stream9: docker.io/library/centos:stream9: not found
> ```

This resolved by adding quay as you recommended.  Thanks for the catch!

**Review by thenets — APPROVED** [review]

**Review by AaronH88 — APPROVED** [review]
lgtm

**Pull request merged**